packages/govuk-frontend/src/govuk/component.mjs

```diff
@@ -0,0 +1,31 @@
+import { GOVUKFrontendSupportError } from './errors/index.mjs'
+
+/**
+ * Base Component class
+ *
+ * Centralises the behaviours shared by our components
+ *
+ * @internal
+ * @abstract
+ */
+export class Component {
+  /**
+   * Constructs a new component, validating that GOV.UK Frontend is supported
+   *
+   * @internal
+   */
+  constructor () {
+    this.validateGOVUKFrontendSupport()
```

Comment:
Idea is to keep that constructor a series of method calls to give an easy overview of what's happening. It'll also allow fine grained overriding should someone need to reach to the internals to amend one of these behaviours 😊